core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-add-variable/services/dot-fields.service.ts

```diff
@@ -0,0 +1,270 @@
+import { Injectable, inject } from '@angular/core';
+
+import { DotMessageService } from '@dotcms/data-access';
+
+import { FieldTypeWithExtraFields, DotFieldContent } from '../dot-add-variable.models';
+
+@Injectable()
+export class DotFieldsService {
+    private dotMessage = inject(DotMessageService);
+
+    // You can add here a new variable and add the custom code that it has
+    private readonly getCodeTemplate: Record<string, (variable: string) => string> = {
+        image: (variable) =>
+            `#if ($UtilMethods.isSet(\${${variable}ImageURI}))\n    <img src="$!{dotContentMap.${variable}ImageURI}" alt="$!{dotContentMap.${variable}ImageTitle}" />\n#end`,
+        file: (variable) =>
+            `#if (\${${variable}FileURI})\n    <a href="$!{dotContentMap.${variable}FileURI}">$!{dotContentMap.${variable}FileTitle}</a>\n#end`,
+        binaryFile: (variable) =>
+            `#if ($UtilMethods.isSet(\${${variable}BinaryFileURI}))\n    <a href="$!{dotContentMap.${variable}BinaryFileURI}?force_download=1&filename=$!{dotContentMap.${variable}BinaryFileTitle}">$!{dotContentMap.${variable}BinaryFileTitle}</a>\n#end`,
+        binaryResized: (variable) =>
+            `#if ($UtilMethods.isSet(\${${variable}BinaryFileURI}))\n    <img src="/contentAsset/resize-image/\${ContentIdentifier}/${variable}?w=150&h=100&language_id=\${language}" />\n#end`,
+        binaryThumbnailed: (variable) =>
+            `#if ($UtilMethods.isSet(\${${variable}BinaryFileURI}))\n    <img src="/contentAsset/image-thumbnail/\${ContentIdentifier}/${variable}?w=150&h=150&language_id=\${language}" />\n#end`,
+        blockEditor: (variable) => `$dotContentMap.get('${variable}').toHtml()`,
+        default: (variable) => `$!{dotContentMap.${variable}}`
+    };
+
+    readonly contentIdentifierField: DotFieldContent = {
+        name: this.dotMessage.get('Content-Identifier-value'),
+        variable: 'ContentIdentifier',
+        fieldTypeLabel: this.dotMessage.get('Content-Identifier'),
+        codeTemplate: this.getCodeTemplate.default('ContentIdentifier')
+    };
+
+    // You can add here a new fieldType and add the meta data fields that it has
+    readonly fields: Record<
+        FieldTypeWithExtraFields, // Remember to add the new fieldType here
+        (variableContent: DotFieldContent) => DotFieldContent[]
+    > = {
+        Image: ({ variable, name, fieldTypeLabel }) => [
```

Comment:
To improve the readability and maintainability of this object, it would be nice to create private functions for each field such as `getImageField`, `getBinaryField`, and so on. This approach will help in keeping the object's codebase more organized and easier to comprehend.

Reply:
I like the idea, I will implement it

Reply:
Done!